### PR TITLE
EZP-27888: Simplified eZPlatformAdminUI code and cleaned structure

### PR DESCRIPTION
**Pull request by Nattfarinn**
Jira ticket related: https://jira.ez.no/browse/EZP-27888
----

Don't even try to review it as one big change. I did everything I could to split it into separate, single responsibility commits. It's not perfect, there are probably parts that should not belong to this specific commit, but it should be at least readable.

- EZP-27888 tickets are mostly related to cleaning the code from obsolete stuff
- EZP-27935 tickets are related to unification of how Symfony Forms are used on our side
- EZP-28125 tickets are related to missing translations and unification of Notifications

### CODE REVIEW

File: src/lib/Form/SubmitHandler.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+     *
+     * @return null|Response
+     */
+    public function handle(FormInterface $form, callable $handler): ?Response
+    {
+        $data = $form->getData();
+
+        if ($form->isValid()) {
+            try {
+                $result = call_user_func_array($handler, [$data]);
```
Comment (mikadamczyk) [inline src/lib/Form/SubmitHandler.php]: Could be changed to `$result = $handler($data);`. I think it's more readable

File: src/lib/Form/SubmitHandler.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+     *
+     * Useful for dynamic URLs based on the result of submit.
+     * Params to generate route are returned from callable(mixed):array submit handler.
+     *
+     * @param string|null $field
+     * @param array $params
+     *
+     * @return null|RedirectResponse
+     */
+    protected function getRedirection(string $field = null, $params = []): ?RedirectResponse
```
Comment (mikadamczyk) [inline src/lib/Form/SubmitHandler.php]: I think parameter $params can be declered as array

File: src/lib/Form/SubmitHandler.php (hunk 1 above)
Comment (Nattfarinn) [reply src/lib/Form/SubmitHandler.php]: @webhdx . Told you :P

File: src/bundle/Controller/ContentTypeGroupController.php [hunk 3]
```diff
@@ -8,89 +8,152 @@
...
+        FormFactory $formFactory,
+        SubmitHandler $submitHandler,
+        array $languages
+    ) {
+        $this->notificationHandler = $notificationHandler;
+        $this->translator = $translator;
+        $this->contentTypeService = $contentTypeService;
+        $this->formFactory = $formFactory;
+        $this->submitHandler = $submitHandler;
+        $this->languages = $languages;
```
Comment (mikadamczyk) [inline src/bundle/Controller/ContentTypeGroupController.php]: $this->languages is not used

File: src/bundle/Controller/RoleAssignmentController.php [hunk 4]
```diff
@@ -16,40 +16,71 @@
...
+        TranslatorInterface $translator,
+        RoleService $roleService,
+        SearchService $searchService,
+        FormFactory $formFactory,
+        SubmitHandler $submitHandler
+    ) {
+        $this->notificationHandler = $notificationHandler;
+        $this->translator = $translator;
         $this->roleService = $roleService;
         $this->searchService = $searchService;
```
Comment (mikadamczyk) [inline src/bundle/Controller/RoleAssignmentController.php]: `$this->searchService` is not used

File: src/lib/Form/SubmitHandler.php (hunk 1 above)
Comment (webhdx) [reply src/lib/Form/SubmitHandler.php]: I'm used to seeing `call_user_func_array()` more frequently. But up to you :P

File: src/bundle/Controller/SectionController.php [hunk 5]
```diff
@@ -43,28 +46,37 @@ class SectionController extends Controller
...
         FormFactory $formFactory,
         SectionCreateMapper $sectionCreateMapper,
-        SectionUpdateMapper $sectionUpdateMapper
+        SectionUpdateMapper $sectionUpdateMapper,
+        SubmitHandler $submitHandler
     ) {
+        $this->notificationHandler = $notificationHandler;
+        $this->translator = $translator;
         $this->sectionService = $sectionService;
         $this->searchService = $searchService;
```
Comment (mikadamczyk) [inline src/bundle/Controller/SectionController.php]: $this->searchService is not used

**Comment by m-tyrala** [comment]
There's a problem in this branch that not occurs in master v1 or v2 - when you add one of "All functions" policies to role, then the list of policies along with "Add new policy" button disappear from role view.

**Comment by webhdx** [comment]
And there is that bug in Versions tab, twig files were not fixed after forms refactoring.

**Comment by barbaragr** [comment]
There is a problem, when you add new content type and next you want to edit it, after clicking the button Edit, you receive 404 error. Preview works fine.

File: src/bundle/Resources/views/admin/content_type_group/add.html.twig [hunk 6]
```diff
@@ -4,15 +4,15 @@
 
 {% block breadcrumbs %}
     {% include '@EzPlatformAdminUi/parts/breadcrumbs.html.twig' with { items: [
-        { value: 'breadcrumb.admin'|trans([], 'messages') },
-        { url: path('ezplatform.content_type_group.list'), value: 'content_type_group.breadcrumb.list'|trans },
-        { value: 'content_type_group.breadcrumb.add'|trans }
+        { value: 'breadcrumb.admin'|trans(domain='messages')|desc('Admin') },
```
[the remark this answers is not in the document]
Comment (Nattfarinn) [reply src/bundle/Resources/views/admin/content_type_group/add.html.twig]: > I'm not a fan of named parameters at it looks bad to the eye, also I haven't seen it in examples in Symfony's doc.

I beg to differ. What looks better? `trans([], 'messages')` or `trans(domain='messages')`

It's in Twig documentation: https://twig.symfony.com/doc/2.x/templates.html#named-arguments

**Comment by m-tyrala** [comment]
EZP-28133 and EZP-28132 - test ok
EZP-28144 - test failed, reopen

In addition to that, I'm not able to add field to Content Type - "The CSRF token is invalid. Please try to resubmit the form" :
1. Go to 'Admin / Content Types / Content'
2. Click 'Edit' right to content type, e.z. Article
3. Click 'Add field definition'

Refreshing or even relogging does not help.

**Pull request merged**